mteb/models/dummy_retrieval_pipeline.py

```diff
@@ -0,0 +1,182 @@
+            **kwargs,
+        )
+        self.mdl.eval()
+
+        # Load processor
+        self.processor = processor_class.from_pretrained(model_name)
+
+    def encode(self, sentences, **kwargs):
+        return self.get_text_embeddings(texts=sentences, **kwargs)
+
+    def encode_input(self, inputs):
+        return self.mdl(**inputs)
+
+    def get_image_embeddings(
+        self,
+        images,
+        batch_size: int = 32,
+        **kwargs,
+    ):
+        import torchvision.transforms.functional as F
+
+        all_embeds = []
+
+        if isinstance(images, DataLoader):
+            iterator = images
+        else:
+            iterator = DataLoader(images, batch_size=batch_size)
+
+        with torch.no_grad():
+            for batch in tqdm(iterator):
+                # batch may be list of tensors or PIL
+                imgs = [
+                    F.to_pil_image(b.to("cpu")) if not isinstance(b, Image.Image) else b
+                    for b in batch
+                ]
+                inputs = self.processor.process_images(imgs).to(self.device)
+                outs = self.encode_input(inputs)
+                all_embeds.extend(outs.cpu().to(torch.float32))
+
+        padded = torch.nn.utils.rnn.pad_sequence(
+            all_embeds, batch_first=True, padding_value=0
+        )
+        return padded
+
+    def get_text_embeddings(
+        self,
+        texts,
+        batch_size: int = 32,
+        **kwargs,
+    ):
+        all_embeds = []
+        with torch.no_grad():
+            for i in tqdm(range(0, len(texts), batch_size)):
+                batch = [
+                    self.processor.query_prefix
+                    + t
+                    + self.processor.query_augmentation_token * 10
+                    for t in texts[i : i + batch_size]
+                ]
+                inputs = self.processor.process_texts(batch).to(self.device)
+                outs = self.encode_input(inputs)
+                all_embeds.extend(outs.cpu().to(torch.float32))
+
+        padded = torch.nn.utils.rnn.pad_sequence(
+            all_embeds, batch_first=True, padding_value=0
+        )
+        return padded
+
+    def get_fused_embeddings(
+        self,
+        texts: list[str] | None = None,
+        images: list[Image.Image] | DataLoader | None = None,
+        *,
+        task_name: str | None = None,
+        prompt_type: PromptType | None = None,
+        batch_size: int = 32,
+        fusion_mode="sum",
+        **kwargs: Any,
+    ):
+        raise NotImplementedError(
+            "Fused embeddings are not supported yet. Please use get_text_embeddings or get_image_embeddings."
+        )
+
+    def calculate_probs(self, text_embeddings, image_embeddings):
+        scores = self.similarity(text_embeddings, image_embeddings).T
+        return scores.softmax(dim=-1)
+
+    def similarity(self, a, b):
+        return self.processor.score(a, b, device=self.device)
+
+
+class PipelineColPaliWrapper:
```

Comment:
I think we could create a `RetrievalPipeline` abstract class,  initially with a single abstract method `pipeline_search()`, then all retrieval pipelines should inherit from that class.
Then this dummy class would extend it, for example `PipelineColPaliWrapper(RetrievalPipeline)`